pkg/operator/ceph/file/controller.go

```diff
@@ -255,9 +255,13 @@ func (r *ReconcileCephFilesystem) reconcileCreateFilesystem(cephFilesystem *ceph
 		return reconcile.Result{}, errors.Wrapf(err, "failed to get controller %q owner reference", cephFilesystem.Name)
 	}
 
-	// preservePoolsOnDelete being set to true does not make sense because of data-loss concerns (see #6492).
+	// preservePoolsOnDelete being set to true has data-loss concerns and is deprecated (see #6492).
+	// If preservePoolsOnDelete is set to true, assume the user means preserveFilesystemOnDelete instead.
 	if cephFilesystem.Spec.PreservePoolsOnDelete {
-		return reconcile.Result{}, errors.New("preservePoolsOnDelete has been deprecated. Set preserveFilesystemOnDelete instead")
+		if !cephFilesystem.Spec.PreserveFilesystemOnDelete {
+			logger.Warningf("preservePoolsOnDelete (currently set 'true') has been deprecated in favor of preserveFilesystemOnDelete (currently set 'false') due to data loss concerns (see rook/rook#6492); Rook will assume preserveFilesystemOnDelete 'true'")
```

Comment:
My preference is not to reference a specific issue unless perhaps there is a follow-up item for maintainers, but users don't need to know the issue numbers.
```suggestion
			logger.Warningf("preservePoolsOnDelete (currently set 'true') has been deprecated in favor of preserveFilesystemOnDelete (currently set 'false') due to data loss concerns so Rook will assume preserveFilesystemOnDelete 'true'")
```

Reply:
Agreed, the user should not have to look for the rationale IMO. If they really want too they will have to dig a bit and will eventually find the Rook issue.